### PR TITLE
DOCSP-41129: update docs

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41129>
Staging - https://preview-mongodbrustagir.gatsbyjs.io/kotlin-sync/DOCSP-41129-update/write/update/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST? **not related**
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/update.txt [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+- ``updateOne()``, which updates *the first document* that matches the search criteria
+- ``updateMany()``, which updates *all documents* that match the search criteria
+
+Each update method requires the following parameters:
+
+- **Query filter**, which matches which documents to update. To learn
+  more about query filters, see the :ref:`kotlin-sync-specify-query`
+  guide.
+- **Update document**, which specifies the update operator, or the kind of update to
+  perform, and the fields and values that should change. For a list of update
```
Comment (mcmorisi) [inline source/write/update.txt:58]: There might be a better way to say this in active voice that I'm not thinking of, but to address the Vale flag:

```suggestion
  perform, and the fields and values to be updated. For a list of update
```

File: source/write/update.txt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+The following example uses the ``updateOne()`` method to update the
+``name`` value of a document from ``"Happy Garden"`` to ``"Mountain
+House"``:
+
+.. literalinclude:: /includes/write/update.kt
+   :start-after: start-update-one
+   :end-before: end-update-one
+   :language: kotlin
+   :copyable:
```
Comment (mcmorisi) [inline source/write/update.txt:73]: Formatting tweak:
```suggestion
.. literalinclude:: /includes/write/update.kt
   :start-after: start-update-one
   :end-before: end-update-one
   :language: kotlin
   :copyable:
   :dedent:
```

File: source/write/update.txt [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+The following example uses the ``updateMany()`` method to update all documents
+in which the ``name`` value is ``"Starbucks"``. The update renames the
+``address`` field to ``location``.
+
+.. literalinclude:: /includes/write/update.kt
+   :start-after: start-update-many
+   :end-before: end-update-many
+   :language: kotlin
+   :copyable:
```
Comment (mcmorisi) [inline source/write/update.txt:86]: Same as above
```suggestion
.. literalinclude:: /includes/write/update.kt
   :start-after: start-update-many
   :end-before: end-update-many
   :language: kotlin
   :copyable:
   :dedent:
```

File: source/write/update.txt [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
+
+   * - ``arrayFilters()``
+     - | Contains a list of filters that specifies which array elements an update applies
+         to.
+
+   * - ``hint()``
+     - | Gets or sets the index to scan for documents.
```
Comment (mcmorisi) [inline source/write/update.txt:131]: Q: The intro to this section refers to this collection of methods as "setter methods" but this method in particular mentions getting OR setting. Does it make sense to discuss the get functionality of this method within the context of this table?

**Review by mcmorisi — CHANGES_REQUESTED** [review]
Good stuff! A few open questions and some small suggestions.

File: source/write/update.txt (hunk 2 above)
Comment (rustagir) [reply source/write/update.txt:73]: fixed all - guess i pushed this PR out too quickly yesterday..

File: source/write/update.txt (hunk 4 above)
Comment (rustagir) [reply source/write/update.txt:131]: Ah, copypasta error. This is only a `set` method in this driver, `getHint()` would be the getter. Will review the descriptions.

File: source/write/update.txt [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+       | Defaults to ``false``.
+
+   * - ``collation()``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
+
+   * - ``arrayFilters()``
+     - | Contains a list of filters that specifies which array elements an update applies
+         to.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/write/update.txt:128]: I can clarify.

**Review by mcmorisi — APPROVED** [review]
LGTM!

**Pull request merged**